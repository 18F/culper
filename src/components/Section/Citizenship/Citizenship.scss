@media only screen and (max-width: 1000px) {
	.citizenship-abroad {
		.block {
			display: block;
		}
	}
}