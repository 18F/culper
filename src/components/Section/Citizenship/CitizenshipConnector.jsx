import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import {
  updateApplication,
  reportErrors
} from '@actions/ApplicationActions'

const connectCitizenshipSection = (Component, { section, subsection, store, storeKey }) => {
  class ConnectedCitizenshipSection extends React.Component {
    constructor(props) {
      super(props)

      this.section = section
      this.subsection = subsection
      this.store = store

      this.handleError = this.handleError.bind(this)
      this.handleUpdate = this.handleUpdate.bind(this)
    }

    handleError(value, arr) {
      const action = reportErrors(this.section, this.subsection, arr)
      this.props.dispatch(action)
      return arr
    }

    handleUpdate(field, values) {
      this.props.dispatch(updateApplication(this.store, field, values))
    }

    render () {
      return (
        <Component
          onUpdate={this.handleUpdate}
          onError={this.handleError}
          {...this.props} />
      )
    }
  }

  ConnectedCitizenshipSection.propTypes = {
    Comments: PropTypes.object,
    update: PropTypes.func,
    validator: PropTypes.func,
    dispatch: PropTypes.func, // Passed in via connect (below)
  }

  const mapStateToProps = (state) => {
    const app = state.application || {}
    const citizenship = app.citizenship || {}
    const errors = app.Errors || {}
    const completed = app.Completed || {}

    switch (storeKey) {
      case 'Status':
        return { ...citizenship.Status } || {}

      case 'Multiple':
        return { ...citizenship.Multiple } || {}

      case 'Passports':
        return { ...citizenship.Passports } || {}
      default:
        return {
          Application: app,
          Citizenship: citizenship,
          Status: citizenship.Status || {},
          Multiple: citizenship.Multiple || {},
          Passports: citizenship.Passports || {},
          Errors: errors.citizenship || [],
          Completed: completed.citizenship || []
        }
    }
  }

  return connect(mapStateToProps)(ConnectedCitizenshipSection)
}

export default connectCitizenshipSection
