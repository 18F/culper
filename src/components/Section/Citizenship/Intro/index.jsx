import React from 'react'

import i18n from 'util/i18n'
import { CITIZENSHIP } from 'config/formSections/citizenship'

import SectionIntro from 'components/Section/shared/SectionIntro'

const Intro = () => (
  <SectionIntro
    title={i18n.t('citizenship.intro.title')}
    body={i18n.m('citizenship.intro.body')}
    sectionKey={CITIZENSHIP.key}
  />
)

export default Intro
