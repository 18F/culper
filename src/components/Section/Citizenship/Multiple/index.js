import Multiple from './Multiple'
export default Multiple
