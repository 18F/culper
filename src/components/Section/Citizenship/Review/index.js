import Review from './Review'
export default Review
