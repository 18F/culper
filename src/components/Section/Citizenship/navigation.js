import compact from 'lodash/compact'
import * as validators from '../../../validators/index'

function citizenshipNavigation(formType) {

  return {
    name: 'Citizenship',
    title: 'Citizenship',
    url: 'citizenship',
    store: 'Citizenship',
    showNumber: true,
    locked: validators.formIsLocked,
    subsections: compact([
      {
        exclude: true,
        name: 'Introduction',
        url: 'intro'
      },
      {
        name: 'Citizenship status',
        url: 'status',
        store: 'Status',
        validator: validators.CitizenshipValidator
      },
      {
        name: 'Dual/multiple citizenship',
        url: 'multiple',
        store: 'Multiple',
        validator: {
          85: validators.CitizenshipMultiple85Validator,
          86: validators.CitizenshipMultipleValidator
        }[formType]
      },
      ['86'].indexOf(formType) > -1 && {
        name: 'Foreign passports',
        url: 'passports',
        store: 'Passports',
        validator: validators.CitizenshipPassportsValidator
      },
      {
        exclude: true,
        name: 'Review',
        url: 'review'
      }
    ])
  }
}

export default citizenshipNavigation
