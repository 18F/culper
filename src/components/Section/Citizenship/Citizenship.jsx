import React from 'react'
import { connect } from 'react-redux'
import { i18n } from '../../../config'
import SectionElement from '../SectionElement'
import SectionComments from '../SectionComments'
import { SectionViews, SectionView } from '../SectionView'
import AuthenticatedView from '../../../views/AuthenticatedView'
import { Field } from '../../Form'
import Status from './Status'
import Multiple from './Multiple'
import Passports from './Multiple/Passports'

class Citizenship extends SectionElement {
  render () {
    return (
      <div>
        <SectionViews current={this.props.subsection} dispatch={this.props.dispatch} update={this.props.update}>
          <SectionView name="intro"
                       back="history/review"
                       backLabel={i18n.t('history.destination.review')}
                       next="citizenship/status"
                       nextLabel={i18n.t('citizenship.destination.status')}>
            <div>
              <Field title={i18n.t('citizenship.intro.title')}
                     titleSize="h2"
                     optional={true}
                     className="no-margin-bottom">
                {i18n.m('citizenship.intro.body')}
              </Field>
            </div>
          </SectionView>

          <SectionView name="review"
                       title={i18n.t('review.title')}
                       para={i18n.m('review.para')}
                       showTop={true}
                       back="citizenship/passports"
                       backLabel={i18n.t('citizenship.destination.passports')}
                       next="military/intro"
                       nextLabel={i18n.t('military.destination.intro')}>
            <Status name="status"
                    {...this.props.Status}
                    section="citizenship"
                    subsection="status"
                    defaultState={false}
                    dispatch={this.props.dispatch}
                    onUpdate={this.handleUpdate.bind(this, 'Status')}
                    onError={this.handleError}
                    required={true}
                    scrollIntoView={false}
                    />
            <hr className="section-divider"/>
            <Multiple name="multiple"
                      {...this.props.Multiple}
                      section="citizenship"
                      subsection="multiple"
                      defaultState={false}
                      dispatch={this.props.dispatch}
                      onUpdate={this.handleUpdate.bind(this, 'Multiple')}
                      onError={this.handleError}
                      required={true}
                      scrollIntoView={false}
                      />
            <hr className="section-divider"/>
            <Passports name="passports"
                       {...this.props.Passports}
                       section="citizenship"
                       subsection="passports"
                       defaultState={false}
                       dispatch={this.props.dispatch}
                       onUpdate={this.handleUpdate.bind(this, 'Passports')}
                       onError={this.handleError}
                       required={true}
                       scrollIntoView={false}
                       />
            <hr className="section-divider" />
            <SectionComments name="comments"
                             {...this.props.Comments}
                             title={i18n.t('citizenship.review.comments')}
                             dispatch={this.props.dispatch}
                             onUpdate={this.handleUpdate.bind(this, 'Comments')}
                             onError={this.handleError}
                             required={false}
                             scrollIntoView={false}
                             />
          </SectionView>

          <SectionView name="status"
                       back="citizenship/intro"
                       backLabel={i18n.t('citizenship.destination.intro')}
                       next="citizenship/multiple"
                       nextLabel={i18n.t('citizenship.destination.multiple')}>
            <Status name="status"
                    {...this.props.Status}
                    dispatch={this.props.dispatch}
                    onUpdate={this.handleUpdate.bind(this, 'Status')}
                    onError={this.handleError}
                    />
          </SectionView>

          <SectionView name="multiple"
                       back="citizenship/status"
                       backLabel={i18n.t('citizenship.destination.status')}
                       next="citizenship/passports"
                       nextLabel={i18n.t('citizenship.destination.passports')}>
            <Multiple name="multiple"
                      {...this.props.Multiple}
                      dispatch={this.props.dispatch}
                      onUpdate={this.handleUpdate.bind(this, 'Multiple')}
                      onError={this.handleError}
                      scrollToBottom={this.props.scrollToBottom}
                      />
          </SectionView>
          <SectionView name="passports"
                       back="citizenship/multiple"
                       backLabel={i18n.t('citizenship.destination.multiple')}
                       next="citizenship/review"
                       nextLabel={i18n.t('citizenship.destination.review')}>
            <Passports name="passports"
                       {...this.props.Passports}
                       dispatch={this.props.dispatch}
                       onUpdate={this.handleUpdate.bind(this, 'Passports')}
                       onError={this.handleError}
                       scrollToBottom={this.props.scrollToBottom}
                       />
          </SectionView>
        </SectionViews>
      </div>
    )
  }
}

function mapStateToProps (state) {
  let app = state.application || {}
  let citizenship = app.Citizenship || {}
  let errors = app.Errors || {}
  let completed = app.Completed || {}
  return {
    Application: app,
    Citizenship: citizenship,
    Status: citizenship.Status || {},
    Multiple: citizenship.Multiple || {},
    Passports: citizenship.Passports || {},
    Comments: citizenship.Comments || {},
    Errors: errors.citizenship || [],
    Completed: completed.citizenship || []
  }
}

Citizenship.defaultProps = {
  section: 'citizenship',
  store: 'Citizenship',
  scrollToBottom: SectionView.BottomButtonsSelector
}

export class CitizenshipSections extends React.Component {
  render () {
    return (
      <div>
        <Status name="status"
                {...this.props.Status}
                defaultState={false}
                dispatch={this.props.dispatch}
                onError={this.props.onError}
                required={true}
                scrollIntoView={false}
                />

        <hr className="section-divider"/>
        <Multiple name="multiple"
                  {...this.props.Multiple}
                  defaultState={false}
                  dispatch={this.props.dispatch}
                  onError={this.props.onError}
                  required={true}
                  scrollIntoView={false}
                  />

        <hr className="section-divider"/>
        <Passports name="passports"
                   {...this.props.Passports}
                   defaultState={false}
                   dispatch={this.props.dispatch}
                   onError={this.props.onError}
                   required={true}
                   scrollIntoView={false}
                   />
        <hr className="section-divider" />
        <SectionComments name="comments"
                         {...this.props.Comments}
                         title={i18n.t('citizenship.review.comments')}
                         dispatch={this.props.dispatch}
                         onError={this.props.onError}
                         required={false}
                         scrollIntoView={false}
                         />
      </div>
    )
  }
}

export default connect(mapStateToProps)(AuthenticatedView(Citizenship))
