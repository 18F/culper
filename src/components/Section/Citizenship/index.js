import Citizenship from './Citizenship'
export default Citizenship
