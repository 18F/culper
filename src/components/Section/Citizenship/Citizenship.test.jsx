import React from 'react'
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import Citizenship from 'components/Section/Citizenship'

describe('The citizenship section', () => {
  const mockStore = configureMockStore()

  it('can review all subsections', () => {
    const store = mockStore({
      application: {
        Settings: {
          formType: 'SF86',
        },
      },
    })
    const component = mount(
      <MemoryRouter>
        <Provider store={store}>
          <Citizenship subsection="review" />
        </Provider>
      </MemoryRouter>
    )
    expect(component.find('div').length).toBeGreaterThan(0)
  })

  it('can go to each subsection', () => {
    const sections = ['status', 'multiple']
    const store = mockStore({
      application: {
        Settings: {
          formType: 'SF86',
        },
      },
    })

    sections.forEach((section) => {
      const component = mount(
        <MemoryRouter>
          <Provider store={store}>
            <Citizenship subsection={section} />
          </Provider>
        </MemoryRouter>
      )
      expect(component.find('div').length).toBeGreaterThan(0)
    })
  })
})
