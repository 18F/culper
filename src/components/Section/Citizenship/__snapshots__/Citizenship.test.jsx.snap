// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The citizenship section renders the CitizenshipSections component 1`] = `
<div>
  <div
    className="section-content status"
    data-section="citizenship"
    data-subsection="status"
  >
    <div
      aria-label="Citizenship status"
      className="field   no-margin-bottom"
      data-uuid="field-MOCK-GUID"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID"
        name="field-MOCK-GUID"
      />
      <h2
        className="title h2"
      >
        Citizenship status
      </h2>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        />
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component"
          />
        </span>
      </div>
    </div>
    <div
      aria-label="Provide your current citizenship status"
      className="field required"
      data-uuid="field-MOCK-GUID"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID"
        name="field-MOCK-GUID"
      />
      <h3
        className="title h3"
      >
        Provide your current citizenship status
      </h3>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        >
          <div
            aria-live="polite"
            className="message error usa-alert usa-alert-error"
            role="alert"
          >
            <div
              className="usa-alert-body"
            >
              <div
                data-i18n="error.required"
              >
                <h5
                  className="usa-alert-heading"
                >
                  Your response is required
                </h5>
                
                
              </div>
            </div>
          </div>
        </span>
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component"
          >
            <div
              className="blocks citizenship-status usa-input-error"
            >
              <div
                className="citizenship-status-citizen block"
              >
                <label
                  className=""
                  htmlFor="citizenship-status-citizen-MOCK-GUID"
                >
                  <input
                    aria-label="[object Object] for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="citizenship-status-citizen-MOCK-GUID"
                    name="citizenship-status-citizen-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="Citizen"
                  />
                  <span>
                    <div>
                      <p>
                        I am a U.S. citizen or national by birth in the U.S. or U.S. territory/commonwealth
                      </p>
                    </div>
                  </span>
                </label>
              </div>
              <div
                className="citizenship-status-foreignborn block"
              >
                <label
                  className=""
                  htmlFor="citizenship-status-foreignborn-MOCK-GUID"
                >
                  <input
                    aria-label="[object Object] for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="citizenship-status-foreignborn-MOCK-GUID"
                    name="citizenship-status-foreignborn-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="ForeignBorn"
                  />
                  <span>
                    <div>
                      <p>
                        I am a U.S. citizen or national by birth, born to U.S. parent(s), in a foreign country
                      </p>
                    </div>
                  </span>
                </label>
              </div>
              <div
                className="citizenship-status-naturalized block"
              >
                <label
                  className=""
                  htmlFor="citizenship-status-naturalized-MOCK-GUID"
                >
                  <input
                    aria-label="[object Object] for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="citizenship-status-naturalized-MOCK-GUID"
                    name="citizenship-status-naturalized-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="Naturalized"
                  />
                  <span>
                    <div>
                      <p>
                        I am a naturalized U.S. citizen
                      </p>
                    </div>
                  </span>
                </label>
              </div>
              <div
                className="citizenship-status-derived block"
              >
                <label
                  className=""
                  htmlFor="citizenship-status-derived-MOCK-GUID"
                >
                  <input
                    aria-label="[object Object] for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="citizenship-status-derived-MOCK-GUID"
                    name="citizenship-status-derived-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="Derived"
                  />
                  <span>
                    <div>
                      <p>
                        I am a derived U.S. citizen
                      </p>
                    </div>
                  </span>
                </label>
              </div>
              <div
                className="citizenship-status-notcitizen block"
              >
                <label
                  className=""
                  htmlFor="citizenship-status-notcitizen-MOCK-GUID"
                >
                  <input
                    aria-label="[object Object] for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="citizenship-status-notcitizen-MOCK-GUID"
                    name="citizenship-status-notcitizen-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="NotCitizen"
                  />
                  <span>
                    <div>
                      <p>
                        I am not a U.S. citizen
                      </p>
                    </div>
                  </span>
                </label>
              </div>
            </div>
          </span>
        </span>
      </div>
    </div>
  </div>
  <hr
    className="section-divider"
  />
  <div
    className="section-content multiple"
    data-section="citizenship"
    data-subsection="multiple"
  >
    <div
      aria-label="Dual/multiple citizenship"
      className="field   no-margin-bottom"
      data-uuid="field-MOCK-GUID"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID"
        name="field-MOCK-GUID"
      />
      <h2
        className="title h2"
      >
        Dual/multiple citizenship
      </h2>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        />
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component"
          />
        </span>
      </div>
    </div>
    <div
      aria-label="Do you now or have you EVER held dual/multiple citizenships?"
      className="field required  branch has-multiple"
      data-uuid="field-MOCK-GUID"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID"
        name="field-MOCK-GUID"
      />
      <h3
        className="title h3"
      >
        Do you now or have you EVER held dual/multiple citizenships?
      </h3>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        >
          <div
            aria-live="polite"
            className="message error usa-alert usa-alert-error"
            role="alert"
          >
            <div
              className="usa-alert-body"
            >
              <div
                data-i18n="error.required"
              >
                <h5
                  className="usa-alert-heading"
                >
                  Your response is required
                </h5>
                
                
              </div>
            </div>
          </div>
        </span>
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component shrink"
          >
            <div
              className="content"
            />
            <div
              className="blocks option-list branch usa-input-error"
            >
              <div
                className="yes block"
              >
                <label
                  className=""
                  htmlFor="has_multiple-MOCK-GUID"
                >
                  <input
                    aria-label="Yes for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="has_multiple-MOCK-GUID"
                    name="has_multiple-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="Yes"
                  />
                  <span>
                    Yes
                  </span>
                </label>
              </div>
              <div
                className="no block"
              >
                <label
                  className=""
                  htmlFor="has_multiple-MOCK-GUID"
                >
                  <input
                    aria-label="No for null"
                    checked={false}
                    className="usa-input-success"
                    disabled={false}
                    id="has_multiple-MOCK-GUID"
                    name="has_multiple-MOCK-GUID"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="radio"
                    value="No"
                  />
                  <span>
                    No
                  </span>
                </label>
              </div>
            </div>
          </span>
        </span>
      </div>
    </div>
  </div>
  <hr
    className="section-divider"
  />
  <div
    className="section-content passports"
    data-section="citizenship"
    data-subsection="passports"
  >
    <div
      aria-label="Foreign passport information"
      className="field   no-margin-bottom"
      data-uuid="field-MOCK-GUID"
    >
      <a
        aria-hidden="true"
        className="anchor"
        id="field-MOCK-GUID"
        name="field-MOCK-GUID"
      />
      <h2
        className="title h2"
      >
        Foreign passport information
      </h2>
      <span
        className="icon"
      />
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages help-messages"
        />
      </div>
      <div
        className="table expand"
      >
        <span
          aria-live="polite"
          className="messages error-messages"
          role="alert"
        />
      </div>
      <div
        className="table"
      >
        <span
          className="content"
        >
          <span
            className="component"
          />
        </span>
      </div>
    </div>
    <div
      className="has-foreignpassport"
    >
      <div>
        <div
          aria-label="Have you EVER been issued a passport (or identity card for travel) by a country other than the U.S.?"
          className="field required  branch"
          data-uuid="field-MOCK-GUID"
        >
          <a
            aria-hidden="true"
            className="anchor"
            id="field-MOCK-GUID"
            name="field-MOCK-GUID"
          />
          <h3
            className="title h3"
          >
            Have you EVER been issued a passport (or identity card for travel) by a country other than the U.S.?
          </h3>
          <span
            className="icon"
          />
          <div
            className="table expand"
          >
            <span
              aria-live="polite"
              className="messages help-messages"
            />
          </div>
          <div
            className="table expand"
          >
            <span
              aria-live="polite"
              className="messages error-messages"
              role="alert"
            >
              <div
                aria-live="polite"
                className="message error usa-alert usa-alert-error"
                role="alert"
              >
                <div
                  className="usa-alert-body"
                >
                  <div
                    data-i18n="error.required"
                  >
                    <h5
                      className="usa-alert-heading"
                    >
                      Your response is required
                    </h5>
                    
                    
                  </div>
                </div>
              </div>
            </span>
          </div>
          <div
            className="table"
          >
            <span
              className="content"
            >
              <span
                className="component shrink"
              >
                <div
                  className="content"
                />
                <div
                  className="blocks option-list branch usa-input-error"
                >
                  <div
                    className="yes block"
                  >
                    <label
                      className=""
                      htmlFor="branchcollection-MOCK-GUID"
                    >
                      <input
                        aria-label="Yes for null"
                        checked={false}
                        className="usa-input-success"
                        disabled={false}
                        id="branchcollection-MOCK-GUID"
                        name="branchcollection-MOCK-GUID"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        type="radio"
                        value="Yes"
                      />
                      <span>
                        Yes
                      </span>
                    </label>
                  </div>
                  <div
                    className="no block"
                  >
                    <label
                      className=""
                      htmlFor="branchcollection-MOCK-GUID"
                    >
                      <input
                        aria-label="No for null"
                        checked={false}
                        className="usa-input-success"
                        disabled={false}
                        id="branchcollection-MOCK-GUID"
                        name="branchcollection-MOCK-GUID"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        type="radio"
                        value="No"
                      />
                      <span>
                        No
                      </span>
                    </label>
                  </div>
                </div>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
